system/lib/wasmfs/syscalls.cpp

```diff
@@ -726,6 +726,33 @@ int __syscall_fchdir(int fd) {
   return doChdir(openFile->locked().getFile());
 }
 
+
+const char* __syscall_getcwd_woutargs() {
+
+  auto curr = wasmFS.getCWD();
+
+  std::string result = "";
+
+  while (curr != wasmFS.getRootDirectory()) {
+    auto parent = curr->locked().getParent();
+    auto parentDir = parent->dynCast<Directory>();
+
+    auto name = parentDir->locked().getName(curr);
+    result = '/' + name + result;
+    curr = parentDir;
+  }
+
+  // Check if the cwd is the root directory.
+  if (result.empty()) {
+    result = "/";
+  }
+
+  auto res = result.c_str();
+
+  return res;
```

Comment:
Here you are returning a pointer into `result` which is a std::string on the stack.. so the pointer won't be valid once you return.

Reply:
Also, is there some reason to create a new function that seems to duplicate the existing `__syscall_getcwd` ?

Reply:
@sbc100  I refered the documention which states that the cwd() does not take any arguments so I had to make a new getcwd() function because the old one takes the path and size as the arguments. Do correct  if I am thinking in the wrong direction or if the function can support arguments contrary to what the documentation states.

Kindly refer https://emscripten.org/docs/api_reference/Filesystem-API.html#new-file-system-wasmfs for the FS.cwd() function definition.

Reply:
The high level `FS.cwd()` doesn't take any args but it can call the lower level __syscall_getcwd that does take args.

IIUC it can/should then read the result from the output pointer and return a JS string based on those results.